boards/arm/sam4s_xplained/ext_ram.c

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) Justin Watson 2017
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <init.h>
+#include <nanokernel.h>
+#include <pinmux.h>
+#include <soc.h>
+#include <sys_io.h>
+
+/* Static Memory Controller (SMC) */
```

Comment:
Seems like this is pretty board / memory connected specific.  We could have a helper function that lives in arch/arm/soc/atmel_sam/sam4s/ to do the pin setup & anything generic, but I don't see this being a driver as its seems so trivial & board/memory specific.